lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -307,24 +330,246 @@ class FwProgCmdProvider : public CmdProvider {
                 m_nextCmd = FLSPRG_COMPLETE;
                 while (gapFill > 0U)
                 {
-                    m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH + m_firmwareBytesRead] = 0xFF;
+                    m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH +m_firmwareBytesRead] = 0xFF;   //m_firmwareBuffer[WRITE_BLOCK_CMD_LENGTH + m_firmwareBytesRead] = 0xFF;
                     --gapFill;
                     ++m_firmwareBytesRead;
                 }
             }
             else {
                 m_nextCmd = FLSPRG_SET_ADDR;
             }
-            
             fwProgCmd.command = m_updatedCmdBuffer;
             fwProgCmd.commandsize = sizeof(m_updatedCmdBuffer);
+ 
+            m_count++;
+            LOG_DEBUG("Write count = %d", m_count);
             return true;
         }
 
         return true;
     }
 };
 
+/**
+ * @class ByteCheckCmdProvider
+ * @brief Implementation of a command provider for reading and verifying bytes.
+ *
+ * This class provides functionality to read bytes and compare them against expected bytes.
+ * It is designed to be used in a bootloader communication system to ensure the integrity
+ * of data read from a target device.
+ */
+class ByteCheckCmdProvider : public CmdProvider {
+public:
+    /**
+     * @brief Constructs a new ByteCheckCmdProvider object.
+     *
+     * @param [in] fileName The name of the file containing the expected bytes.
+     * @remarks The file specified by fileName should contain the expected bytes to be compared with.
+     */
+    ByteCheckCmdProvider(const char* fileName) :
+        m_updatedCmd(),
+        m_firmwareBuffer(),
+        m_updatedCmdBuffer{},

```

Comment:
The return of the fimwareBuffer . Why ? I think you need a Responseinfo with 512 byte space, not another buffer.

Reply:
The m_firmwareBuffer is necessary for processing firmware data read from a file. It acts as a buffer to hold the firmware data temporarily, allowing for preprocessing before further use.

On the other hand, m_updatedRsp serves as an intermediate storage for expected responses. It contains fields to hold the expected response data, which is derived from the firmware data stored in m_firmwareBuffer.

Together, m_firmwareBuffer and m_updatedRsp facilitate the preprocessing and dynamic updating of expected response data based on the firmware data. The rsp pointer provides access to the updated response data. 
FLASH_BLOCK_LENGTH is 128bytes . It will be enough to store the data.